ENIGMAsystem/SHELL/Platforms/Win32/WINDOWSmain.cpp

```diff
@@ -498,9 +497,16 @@ void execute_program(std::string operation, std::string fname, std::string args,
 
   //wait until a file is finished printing
   if (wait && lpExecInfo.hProcess != NULL) {
-    ::WaitForSingleObject(lpExecInfo.hProcess, INFINITE);
-    ::CloseHandle(lpExecInfo.hProcess);
+    while (WaitForSingleObject(lpExecInfo.hProcess, 0) == WAIT_TIMEOUT) {
```

Comment:
This 0 causes the loop to busy-wait. Better to use 1, or even a larger value like 5.

Reply:
I disagree; doing that will make the window cursor turn into a loading cursor, and to me that is very undesirable behavior, and that is the only difference it made with my testing. The way I have is will allow you to click on the close button and will act like the button is clicked, but the game will not close because the process is waiting. This is what it does on Linux and is the correct behavior. I encourage you to actually test this before you conclude it's not correct.

Reply:
@JoshDreamland does this apply to this function only, or does it work differently when using a thread handle, like i did in execute_shell_for_ouput()?

Reply:
Basically, anywhere a loop runs without sleeping, it will tie up a CPU thread. So if this code appears elsewhere and I missed it, it'll have the same problem. You can fix it by specifying a higher timeout (but not so high that the operating system will assume you've died), or by adding a call to `sleep`.